csp/impl/mem_cache.py

```diff
@@ -204,10 +198,8 @@ def __call__(*args, **kwargs):
                 except TypeError as e:
                     if force_memoize:
                         raise
-                    if not warned_flag.value:
-                        logging_context = function_name if function_name else str(func)
-                        logging.debug(f"Not memoizing output of {str(logging_context)}: {str(e)}")
-                        warned_flag.value = True
+                    logging_context = function_name if function_name else str(func)
+                    warn(f"Not memoizing output of {str(logging_context)}: {str(e)}", Warning)
```

Comment:
will warn ensure this warning is only emitted once?  Even if it does though this seems to change behavior, since the old code seems like it would warn once per graph method

Reply:
Yes, by default `warn` will only warn once, however a user can change the behavior by explicitly calling `warnings.simplefilter('always', Warning)`. Since a unique warning _message_ will trigger a new warning, I don't see the need of warning once per graph method - really we should be warning once per "non-memoizable" type, which this new approach achieves.

```
>>> from warnings import warn
>>> strs = ('a', 'b', 'a')
>>> for s in strs:
...     warn(s, Warning)
... 
<stdin>:2: Warning: a
<stdin>:2: Warning: b
>>>
```